### PR TITLE
Add rate limiting for snapshot calls #58

### LINKED ISSUE #58: Add rate limiting to TakeSnapshot d-bus call
If TakeSnapshot was already called recently then it should return (without error) to prevent unnecessary system load.

### PR DESCRIPTION
**Pull request by rpoletaev**
Closes #58

### CODE REVIEW

File: cmd/thermal-recorder/snapshot.go [hunk 1]
```diff
@@ -79,7 +88,14 @@ func newSnapshot(dir string) error {
 		return err
 	}
 	defer out.Close()
-	return png.Encode(out, g16)
+
+	if err := png.Encode(out, g16); err != nil {
+		return err
+	}
+
+	// the time will be changed only if the attempt is successfull
```
Comment (mjs) [inline cmd/thermal-recorder/snapshot.go]: successful

**Review by mjs — APPROVED** [review]
Wonderful! Thanks!

How were you able to test this?

**Comment by rpoletaev** [comment]
> Wonderful! Thanks!
> 
> How were you able to test this?

I do not tested it in this code. But i made separate example to check this approach.
In order to write a test for this functionality I need to do refactoring so that I can make a stub for functionality that does not apply to my task.

File: cmd/thermal-recorder/snapshot.go (hunk 1 above)
Comment (rpoletaev) [reply cmd/thermal-recorder/snapshot.go]: done

**Comment by mjs** [comment]
> > Wonderful! Thanks!
> > How were you able to test this?
> 
> I do not tested it in this code. But i made separate example to check this approach.
> In order to write a test for this functionality I need to do refactoring so that I can make a stub for functionality that does not apply to my task.

It's ok. I was just curious how you verified that the change worked. What you've done sounds reasonable given the current state of things.

**Pull request merged**